src/main/java/ch/cern/eam/wshub/core/services/workorders/impl/ChecklistServiceImpl.java

```diff
@@ -59,10 +66,202 @@ public ChecklistServiceImpl(ApplicationData applicationData, Tools tools, InforW
+
+		boolean isReviewer = noRequiredQualifications || reviewerQualification == null && (isPerformer1 || isPerformer2)
+							 || qualifications.contains(reviewerQualification);
+
+		if(isPerformer1 || isReviewer){
+			WorkOrderActivityChecklistSignatureResult perf1 = new WorkOrderActivityChecklistSignatureResult();
+			perf1.setType("PB01");
+			perf1.setSigner(workOrderActivityCheckList.getPerformer1Name());
+			perf1.setViewAsPerformer(isPerformer1);
+			perf1.setViewAsReviewer(isReviewer);
+			perf1.setTime(workOrderActivityCheckList.getTimePerf1() != null ? dateAdapter.marshal(workOrderActivityCheckList.getTimePerf1()) : null);
+			perf1.setResponsibilityCode(performer1Qualification);
+			signatures.add(perf1);
+		}
+
+		if(isPerformer2 || isReviewer){
+			WorkOrderActivityChecklistSignatureResult perf2 = new WorkOrderActivityChecklistSignatureResult();
+			perf2.setType("PB02");
+			perf2.setSigner(workOrderActivityCheckList.getPerformer2Name());
+			perf2.setViewAsPerformer(isPerformer2);
+			perf2.setViewAsReviewer(isReviewer);
+			perf2.setTime(workOrderActivityCheckList.getTimePerf2() != null ?
+					dateAdapter.marshal(workOrderActivityCheckList.getTimePerf2()) : null);
+			perf2.setResponsibilityCode(performer2Qualification != null ? performer2Qualification : performer1Qualification);
+			signatures.add(perf2);
+		}
+
+		if(isReviewer){
+			WorkOrderActivityChecklistSignatureResult reviewer = new WorkOrderActivityChecklistSignatureResult();
+			reviewer.setType("RB01");
+			reviewer.setSigner(workOrderActivityCheckList.getReviewerName());
+			reviewer.setViewAsPerformer(workOrderActivityCheckList.getPerformer1Name() != null
+										|| workOrderActivityCheckList.getPerformer2Name() != null);
+			reviewer.setTime(workOrderActivityCheckList.getTimeRev1() != null ? dateAdapter.marshal(workOrderActivityCheckList.getTimeRev1()) : null);
+			if(reviewerQualification == null && performer1Qualification != null && performer2Qualification == null)
+				reviewerQualification = performer1Qualification;
+
+
+			reviewer.setResponsibilityCode(reviewerQualification);
+			signatures.add(reviewer);
+		}
+
+		return signatures.toArray(signatures.toArray(new WorkOrderActivityChecklistSignatureResult[0]));
+	}
+
+	private MP7999_GetWorkOrderActivityCheckListDefault_001_Result getSignatureWS(InforContext context, String workOrderCode, String activityCode) throws InforException {
+		WorkOrderActivityCheckListDefault workOrderActivityCheckListDefault = new WorkOrderActivityCheckListDefault();
+		workOrderActivityCheckListDefault.setActivityCode(new BigInteger(activityCode));
+		workOrderActivityCheckListDefault.setWorkOrderId(workOrderCode);
+
+		MP7999_GetWorkOrderActivityCheckListDefault_001 getWorkOrderActivityCheckListDefault = new MP7999_GetWorkOrderActivityCheckListDefault_001();
+		tools.getInforFieldTools().transformWSHubObject(getWorkOrderActivityCheckListDefault, workOrderActivityCheckListDefault, context);
```

Comment:
Why is workOrderActivityCheckListDefault transformed to an object that is basically the same? Why not set the fields of getWorkOrderActivityCheckListDefault directly?

Reply:
Done